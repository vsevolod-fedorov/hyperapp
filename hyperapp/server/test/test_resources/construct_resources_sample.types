
sample_servant_d = record

sample = record:
  id: int

sample_item = record:
  key: int
  name: string
  description: string
