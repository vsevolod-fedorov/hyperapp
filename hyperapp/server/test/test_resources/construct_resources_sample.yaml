import:
  types:
    construct_resources_sample:
      - sample_item
