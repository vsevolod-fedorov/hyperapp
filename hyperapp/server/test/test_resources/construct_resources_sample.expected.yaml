import:
- legacy_module.construct_resources_sample
- legacy_service.mosaic
- legacy_service.peer_registry
- legacy_service.rpc_call_factory
- legacy_service.web
- legacy_type.attribute.attribute
- legacy_type.builtin.int
- legacy_type.call.call
- legacy_type.command.object_commands_d
- legacy_type.construct_resources_sample.sample
- legacy_type.construct_resources_sample.sample_servant_d
- legacy_type.construct_resources_sample.sample_servant_open_d
- legacy_type.construct_resources_sample.sample_servant_parent_d
- legacy_type.impl.impl_association
- legacy_type.impl.list_impl
- legacy_type.impl.object_command_impl
- legacy_type.lcs.lcs_set_resource_association
- legacy_type.partial.partial
associations:
- _type: legacy_type.impl.impl_association
  piece_t: legacy_type.construct_resources_sample.sample
  implementation: sample_servant_impl
- _type: legacy_type.lcs.lcs_set_resource_association
  dir:
  - sample_servant_d
  - object_commands_d
  value: sample_servant_open_command
- _type: legacy_type.lcs.lcs_set_resource_association
  dir:
  - sample_servant_d
  - object_commands_d
  value: sample_servant_parent_command
definitions:
  object_commands_d:
    _type: legacy_type.call.call
    function: legacy_type.command.object_commands_d
  sample_servant_attribute:
    _type: legacy_type.attribute.attribute
    object: legacy_module.construct_resources_sample
    attr_name: SampleServant
  sample_servant_d:
    _type: legacy_type.call.call
    function: legacy_type.construct_resources_sample.sample_servant_d
  sample_servant_impl:
    _type: legacy_type.impl.list_impl
    function: sample_servant_partial
    key_attribute: key
    key_t: legacy_type.builtin.int
    dir: sample_servant_d
  sample_servant_open_command:
    _type: legacy_type.impl.object_command_impl
    object_ctr: sample_servant_partial
    method: open
    params:
    - current_key
    dir: sample_servant_open_d
  sample_servant_open_d:
    _type: legacy_type.call.call
    function: legacy_type.construct_resources_sample.sample_servant_open_d
  sample_servant_parent_command:
    _type: legacy_type.impl.object_command_impl
    object_ctr: sample_servant_partial
    method: parent
    params: []
    dir: sample_servant_parent_d
  sample_servant_parent_d:
    _type: legacy_type.call.call
    function: legacy_type.construct_resources_sample.sample_servant_parent_d
  sample_servant_partial:
    _type: legacy_type.partial.partial
    function: sample_servant_attribute
    params:
      mosaic: legacy_service.mosaic
      web: legacy_service.web
      peer_registry: legacy_service.peer_registry
      rpc_call_factory: legacy_service.rpc_call_factory
